src/main/java/lesson2/problem6/MergeKLists.java

```diff
@@ -0,0 +1,69 @@
+package lesson2.problem6;
+
+import lesson2.problem0.Link;
+import lesson2.problem0.LinkedList;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.PriorityQueue;
+
+public class MergeKLists {
+
+    public static Link mergeLists(List<Link> input) {
+        if (input == null || input.size() == 0) {
+            throw new IllegalArgumentException();
+        }
+        final PriorityQueue<Link> myQueue = new PriorityQueue<>(input.size(), new Comparator<Link>() {
+            @Override
+            public int compare(Link o1, Link o2) {
+                return 0;
```

Comment:
You should implement this by saying "return o1 - o2". If you return 0, the priority queue will not order the items properly. 0 means that two arguments o1 and o2 are equal.